staging/src/k8s.io/apimachinery/pkg/util/mergepatch/errors.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mergepatch
+
+import (
+	"fmt"
+)
+
+var ErrBadJSONDoc = fmt.Errorf("Invalid JSON document")
```

Comment:
I don't trust the output of fmt.Errorf to always compare correctly (but maybe I'm just paranoid). When I see fmt.Errorf, I think of a class of errors (and one needs to make a separate type to check membership in this class), whereas errors.New is clearly a single error.